### DISCUSSION TITLE
How to isolate tailwind?

### DISCUSSION THREAD

**Discussion by itshallrun** [tailwindlabs/tailwindcss#10751] [Help]
We are currently in the process of gradually migrating a large Reactstrap/Bootstrap-based application to Tailwind. We have tried several approaches, but have encountered issues each time. Either Tailwind breaks the old Bootstrap components, or Bootstrap breaks the new Tailwind components.

The next idea was to isolate Tailwind into a class, so that all components within this class would use Tailwind exclusively. I found a solution [here](https://github.com/tailwindlabs/discuss/issues/294#issuecomment-509605737), but am having trouble making it work.

Do you have any tips on how to set up PostCSS to make the tailwind directives apply only within a class like so?

```
.tailwind {
  @tailwind base;
  @tailwind components;
  @tailwind utilities;
}

```

**Comment by andrepichardo** [comment]
Hi! To avoid the naming conflicts use the [prefix option](https://tailwindcss.com/docs/configuration#prefix) in your tailwind.config.js file

```
// tailwind.config.js
module.exports = {
  prefix: 'tw-',   👈 Use your desired prefix 
}
```

So now you can use the prefix tw- before the class name of tailwind-css which wont break any of your existing styles.

**Note**
If you want tailwind classes to get more precedence than any other css styles , you can set the [important option](https://tailwindcss.com/docs/configuration#important) to true on tailwind.config.js

```
module.exports = {
  important: true,
}
```

Hope this answer helps!

**Comment by itshallrun** [comment]
Hi @andrepichardo and thank you for the suggestion.

we have tried that, but it seems like the tailwind is not properly working without the preflight and/or base. And the css reset is breaking bootstrap. Therefore it seems like this approach is not enough.

In general, is it even possible to nest tailwind like I am trying above?

**Reply by andrepichardo** [reply]
Oh, I see...

Well, according to @adamwathan (creator of Tailwind) in the solution you mentioned [here](https://github.com/tailwindlabs/discuss/issues/294#issuecomment-509605737), there should be no problem to use the next approach:

```
.tailwind {
  @tailwind base;
  @tailwind components;
  @tailwind utilities;
}
```
So yeah, It is possible to nest tailwind like that, assuming there are no import statements in the file and you are using postcss-nested or postcss-nesting to handle nesting.

You can learn more about configuring postcss-nesting [here](https://www.npmjs.com/package/postcss-nesting)!

Alternatively to this, **_as Adam said_**, you could write your own PostCSS plugin that uses comments as markers or something to work around the postcss-import limitation, so you're CSS might look like this:

```
/* start-tailwind-prefix */

/* Tailwind base */
@import './node_module/tailwindcss/base';

/* Custom base */
@import './components/base';

/* Tailwind components */
@import 'tailwindcss/components';

/* Custom components */
@import './components/buttons.css'; 

/* Tailwind utilities */
@import 'tailwindcss/utilities';

/* end-tailwind-prefix */

```

Best of luck!

**Comment by ayophanz** [comment]
Try this  `important: '.review-plugin'` and `corePlugins: {
		preflight: false,
	},`

module.exports = {
	corePlugins: {
		preflight: false,
	},
	important: '.review-plugin',
    content: [
		"./**/*.php",
	    "./*.php",
        "./js/*.js"
    ],
	theme: {
		extend: {},
	},
	plugins: [
        require('@tailwindcss/typography')
   ],
}

**Comment by Maks-T** [comment]
There is the utility
https://www.npmjs.com/package/tailwind-prefix-migrator

**Comment by matthieumiller2101-blip** [comment]
how To build best https
